src/AbstractHydrator.php

```diff
@@ -188,10 +183,13 @@ public function getFilter(): Filter\FilterInterface
      * </code>
      *
      * @param string $name Index in the composite
-     * @param callable|Filter\FilterInterface $filter
+     * @param callable(mixed): mixed|FilterInterface $filter
```

Comment:
```suggestion
     * @param (callable(string, ?object):bool)|FilterInterface $filter
```

Without the parenthesis, the callable is returning `mixed|FilterInterface`

Also, really sorry but I assumed `FilterInterface` here was `Laminas\Filter\FilterInterface`. It is not, and therefore the callable signature should be `callable(string, ?object):bool` as per:

https://github.com/laminas/laminas-hydrator/blob/b45174f0c5ff97c04fea9c120329f8a79e55494f/src/Filter/FilterComposite.php#L156C28-L156C58